NOTICE.md

```diff
@@ -1,13 +1,13 @@
 # Notices for Eclipse Project for JAX-RS
 
-This content is produced and maintained by the Eclipse Project for JAX-RS
+This content is produced and maintained by the **Jakarta RESTful Web Services**
 project.
 
 * Project home: https://projects.eclipse.org/projects/ee4j.jaxrs
 
 ## Trademarks
 
-Eclipse Project for JAX-RS is a trademark of the Eclipse Foundation.
+**Jakarta RESTful Web Services** is a trademark of the Eclipse Foundation.
```

Comment:
I don't know if this is actually a trademark of the Eclipse Foundation, but apparently neither was `Eclipse Project for JAX-RS`... :-)

Reply:
This PR just implements the decisions of our recent discussion. @waynebeaton It would be great if the EF could provide us a list of its trademarks.

Reply:
The Eclipse Foundation claims all project names as trademarks. So... assuming that "Jakarta RESTful Web Services" passes our trademark review (I'll start that soon; it will likely be fine), then the statement will be correct after our restructuring review (#767).

Reply:
@waynebeaton Thanks! :-)